### DISCUSSION TITLE
Creating Quiz Questions

### DISCUSSION THREAD

**Discussion by felkezac** [ucfopen/canvasapi#648] [Q&A] [answered]
I am having trouble creating or editing quiz questions. Regardless what I have tried, it generates a blank question. Attempts to edit the question throw error 500. Am I missing something?

One thing I have tried:
bar = {'question_name': 'asdf', 'question_type': 'numerical_question', 'question_text': 'asdfasdf'}
item = q.create_question(**bar)

**Comment by bennettscience** [comment]
Have you inspected the response from the API? Is it giving an error or returning a properly formatted quiz? A 500 error is something on Instructure's servers and not necessarily related to the library.

**Reply by bennettscience** [reply]
Looking at your attempt, you need to pass the dict as a keyword argument, not as unpacked kwargs. Try calling `q.create_queston(question=bar)` and see if that helps.

**Reply by bennettscience** [reply]
The details about passing keyword arguments to endpoints are here: https://canvasapi.readthedocs.io/en/stable/keyword-args.html

**Comment by bennettscience** [comment, marked as answer]
Posting as a separate answer for clarity. 

The library combines keyword arguments for endpoints which accept different parameters. You should pass dict objects directly to the function either directly or as a variable, like this:

```python
question = {
    "question_name": "Sample question", 
    "question_text": "This is the question text.", 
    "question_type": "numerical_question"
}

q.create_question(question=question)
```

or 

```python
q.create_question(question={
    "question_name": "Sample question", 
    "question_text": "This is the question text.", 
    "question_type": "numerical_question"}
)
```

This is documented, along with other examples, here: https://canvasapi.readthedocs.io/en/stable/keyword-args.html
